widget/splitcontainer.go

```diff
@@ -33,46 +33,24 @@ func NewVSplitContainer(top, bottom fyne.CanvasObject) *SplitContainer {
 
 func newSplitContainer(horizontal bool, leading, trailing fyne.CanvasObject) *SplitContainer {
 	s := &SplitContainer{
+		Offset:     0.5, // Sensible default, can be overriden with SetRatio
 		Horizontal: horizontal,
 		Leading:    leading,
 		Trailing:   trailing,
-		ratio:      0.5, // Sensible default, can be overriden with SetRatio
 	}
 	s.ExtendBaseWidget(s)
 	return s
 }
 
-// SetRatio sets the dividing ratio (0.0 to 1.0) of the SplitContainer.
-// SetRatio ensures neither child is smaller than their min size.
-// SetRatio must be called after Resize to ensure size is set.
+// SetOffset sets the offset (0.0 to 1.0) of the SplitContainer divider.
 // 0.0 - Leading is min size, Trailing uses all remaining space.
 // 0.5 - Leading & Trailing equally share the available space.
 // 1.0 - Trailing is min size, Leading uses all remaining space.
-func (s *SplitContainer) SetRatio(ratio float64) {
-	var max float64
-	var min float64
-	half := float64(halfDividerThickness())
-	if s.Horizontal {
-		sw := float64(s.size.Width)
-		lw := float64(s.Leading.MinSize().Width)
-		tw := float64(s.Trailing.MinSize().Width)
-		max = 1.0 - ((tw + half) / sw)
-		min = (lw + half) / sw
-	} else {
-		sh := float64(s.size.Height)
-		lh := float64(s.Leading.MinSize().Height)
-		th := float64(s.Trailing.MinSize().Height)
-		max = 1.0 - ((th + half) / sh)
-		min = (lh + half) / sh
-	}
-	if ratio < min {
-		ratio = min
-	}
-	if ratio > max {
-		ratio = max
+func (s *SplitContainer) SetOffset(offset float64) {
+	if s.Offset != offset {
```

Comment:
It's cleaner when the main body of a method is it's function - corner cases causing early exit.
So here checking for equality and returning allows the method body to be clearly it's main intent.

Reply:
Done